ocaml/parsing/jane_syntax_parsing.mli

```diff
@@ -118,117 +116,99 @@ module Embedded_name : sig
   val pp_quoted_name : Format.formatter -> t -> unit
 end
 
-(** The type of modules that lift and lower terms from our novel syntactic
-    features from and to an OCaml AST type ([ast]) *)
-module type AST = sig
-  (** The AST type (e.g., [Parsetree.expression]) *)
-  type ast
-
-  (** The "AST description" type, without the location and attributes (e.g.,
-      [Parsetree.expression_desc]) *)
-  type ast_desc
-
-  (** The name for this syntactic category in the plural form; used for error
-      messages (e.g., "expressions") *)
-  val plural : string
+(** An AST desc together with its attributes, including the attributes that
+    signal that the AST should be interpreted as a Jane Syntax AST. These are
+    produced only by [make_jane_syntax] and [make_entire_jane_syntax].
+*)
+module With_attributes : sig
+  type 'desc t = private
+    { jane_syntax_attributes : Parsetree.attributes
+    ; desc : 'desc
+    }
+end
```

Comment:
I strongly dislike that this type escapes into `Jane_syntax` -- I think a downstream user of that module should never have to come here.  Maybe the simplest solution is for `Jane_syntax` to reexport it, though?

Reply:
I'm in favor of re-exporting it from `Jane_syntax` with a more targeted comment. I've done that.